File: tests/fixtures/vulnerable/user_enum.ts
// Vulnerable: differentiated login errors
export async function login(username: string, password: string) {
  const user = await findUser(username);
  if (!user) {
    return { error: "User not found", status: 404 };
  }
  if (!checkPassword(user, password)) {
    return { error: "Wrong password", status: 401 };
  }
  return { token: createToken(user.id), status: 200 };
}

File: tests/fixtures/vulnerable/user_enum.py
"""Vulnerable: differentiated login errors reveal account existence."""


def login(username, password):
    user = find_user(username)
    if not user:
        return {"error": "User not found"}, 404
    if not check_password(user, password):
        return {"error": "Wrong password"}, 401
    return {"token": create_token(user)}, 200
